etc/cmake/options.cmake

```diff
@@ -381,6 +381,13 @@ else()
     message(FATAL_ERROR "Invalid max RCP restoration count: ${OT_RCP_RESTORATION_MAX_COUNT}")
 endif()
 
+option(OT_TCP "enable TCPlp support")
+if(OT_TCP)
+    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_TCP_ENABLE=1")
+else()
+    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_TCP_ENABLE=0")
```

Comment:
While I think such a cmake option works and would exlcude the `tcplp` library from being included, it indirectly adds a restriction in that `TCP` under cmake can now only be enabled using this option.

In OT we have different ways of enabling features and providing the `OPENTHREAD_CONFIG_{}` definitions.
- We can use the build specific option, like cmake options above.
- Or a project can provide a project-specific config header, which includes all `OPENTHREAD_CONFIG_{}` definitions it wants and pass it as config header (e.g. in cmake using `OT_CONFIG`).

The model typically used for other cmake options is that they can enable a feature (set the related `OPENTHREAD_CONFIG_{}` to `1`), but won't explicitly disable it (wont explicitly set it to `0`). 

This way if the cmake option the is not specified/used, it allows the project specific config header to be used and specify the features to enable. 

----

I understand the main goal is to restrict inclusion of `tcplp` in `target_link_libraries`. Here is my suggestion for one possible way to realize this:
- We can add a new cmake option, `OT_EXCLUDE_TCPLP_LIB`
  - If it is used, we then exlude the `tcplp` including in link libs.
  - By default it is not set and therefore we include `tcplp`

By adding a different option for this (and not use `OT_TCP`) we keep these two independent of each other which allow a project-specific config header to be used under `cmake`. 

Thoughts?

Reply:
Thanks for the clarification @abtink. I updated the PR with the suggested `OT_EXCLUDE_TCPLP_LIB` option.

BTW, it seems `OT_ASSERT` doesn't follow the general rule and has to match the `OPENTHREAD_CONFIG_ASSERT_ENABLE` value. Or am I wrong here?

Reply:
Thanks.

Yes. I noticed the same. Some features seem to set the related `OPENTHREAD_CONFIG` explicitly to zero from cmake option. I will submit a PR to change this (soon).